worker/src/core/loaders/message-worker/sms.class.ts

```diff
@@ -60,12 +61,43 @@ class SMS {
       campaignId: number
     }[]
   > {
-    return this.connection
-      .query('SELECT get_messages_to_send_sms(:job_id, :rate);', {
+    const dbResults = await this.connection.query(
+      'SELECT get_messages_to_send_sms(:job_id, :rate);',
+      {
         replacements: { job_id: jobId, rate },
         type: QueryTypes.SELECT,
-      })
-      .then((result) => map(result, 'get_messages_to_send_sms'))
+      }
+    )
+    const result = map(dbResults, 'get_messages_to_send_sms')
+
+    const showContactPref = config.get('phonebookContactPref.enabled')
+    if (showContactPref && result.length > 0) {
+      try {
+        const campaignId = result[0].campaignId as number
+        const emailResult = await this.connection.query<{ email: string }>(
+          'select u.email as email from users u where u.id = (select c.user_id from campaigns c where c.id = :campaignId);',
+          {
+            replacements: { campaignId },
+            type: QueryTypes.SELECT,
+          }
+        )
+        if (!emailResult || emailResult.length === 0) {
+          throw new Error(
+            'Unable to fetch user email from campaign for phonebook contact preference api'
+          )
+        }
+        const userEmail = emailResult[0].email
+        return getMessagesWithContactPrefLinks(result, campaignId, userEmail)
+      } catch (error) {
+        logger.error({
+          message: 'Unable to fetch contact preferences',
+          error,
+          workerId: this.workerId,
+        })
+        // If phonebook is down, we still want to continue sending the messages
+      }
+    }
+    return result
```

Comment:
**[nit]** I think it's clearer to move this `return result` up, i.e.`if (!showContactPref || result.length === 0) return result`. you remove one level of nesting and the code is easier to read